### PR TITLE
Migrating Notion Sync issue Action to Typescript

### PR DESCRIPTION
**Pull request by subhroacharjee**

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -32,7 +32,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        language: [ 'javascript' ]
+        language: [ 'TypeScript' ]
         # CodeQL supports [ 'cpp', 'csharp', 'go', 'java', 'javascript', 'python', 'ruby' ]
         # Learn more about CodeQL language support at https://git.io/codeql-language-support
```
Comment (rlch) [inline .github/workflows/codeql-analysis.yml]: does is support TS?

File: .gitignore [hunk 2]
```diff
@@ -1,14 +1,18 @@
-# Editors
-.vscode/
-.idea/
-*.iml
+# Dependency directory
+node_modules/
+node_modules/**
```
Comment (rlch) [inline .gitignore]: redundant

File: README.md [hunk 3]
```diff
@@ -64,53 +70,36 @@ run()
...
-git checkout -b v1
-git commit -a -m "v1 release"
-```
-
-```bash
-git push origin v1
+$ npm run package
+$ git add dist
+$ git commit -a -m "prod dependencies"
+$ git push origin releases/v1
```
Comment (rlch) [inline README.md]: don't really understand this; can you check over grammar

Also maybe git commit -a -m "chore: Distribute action"

File: README.md [hunk 4]
```diff
@@ -64,53 +70,36 @@ run()
...
 
-## Usage
+## Validate
 
-You can now consume the action by referencing the v1 branch
+You can now validate the action by referencing `./` in a workflow in your repo (see [test.yml](.github/workflows/test.yml))
 
 ```yaml
-uses: actions/javascript-action@v1
+uses: ./
```
Comment (rlch) [inline README.md]: is this correct? shouldn't it be MathGaps/notion-issue-sync@v1?

File: README.md
Comment (rlch) [inline README.md:1]: Check grammar + logic, bit confused - lmk if you'd like to talk this over

File: action.yml [hunk 5]
```diff
@@ -1,15 +1,16 @@
-name: Notion Issue Sync
-description: Invokes a provided Make web-hook for syncing the state of an issue linked to the active PR
+name: 'notion-sync-issue'
+description: 'Provide a description here'
+author: 'subhroacharjee'
 inputs:
   webhook:
-    description: Make webhook for syncing issue state with Notion
     required: true
+    description: 'make integrations webhook'
```
Comment (rlch) [inline action.yml]:
```suggestion
    description: 'Integrations web-hook URL'
```

File: action.yml [hunk 6]
```diff
@@ -1,15 +1,16 @@
...
   token:
-    description: Read-scoped GitHub token
     required: true
+    description: 'private github token'
```
Comment (rlch) [inline action.yml]:
```suggestion
    description: 'Read-scoped GitHub token'
```

File: action.yml [hunk 7]
```diff
@@ -1,15 +1,16 @@
...
   event:
-    description: Should be toJson(github.event) for extracting PR/review state
     required: true
+    description: 'jsonified github event'
```
Comment (rlch) [inline action.yml]: My original description is more robust - how come we changed this?

File: action.yml [hunk 8]
```diff
@@ -1,15 +1,16 @@
-name: Notion Issue Sync
-description: Invokes a provided Make web-hook for syncing the state of an issue linked to the active PR
+name: 'notion-sync-issue'
+description: 'Provide a description here'
+author: 'subhroacharjee'
```
Comment (rlch) [inline action.yml]: What happened here? No need to change any of this metadata. Please revert to original

File: dist/index.js.map
Comment (rlch) [inline dist/index.js.map:1]: dist/ should always be gitignored

**Review by rlch — CHANGES_REQUESTED** [review]

**Comment by rlch** [comment]
I can't see the TypeScript files. Please check your gitignore and do a sanity check of changes (we should never have >10k lines in changes unless bootstrapping a new project, let alone this many files) before creating a PR. If you were bootstrapping a new project, which you kind of are considering you've migrated to TS, i would expect 20 files max.

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (subhroacharjee) [reply .github/workflows/codeql-analysis.yml]: Yup

File: README.md (hunk 4 above)
Comment (subhroacharjee) [reply README.md]: yup this is correct we will use MathGaps/notion-issue-sync in other repositories, given this repository itself is the source for the action we can use this

File: dist/index.js.map
Comment (subhroacharjee) [reply dist/index.js.map:1]: cant really add dist to gitignore given we are using this for the actions

File: .github/workflows/test.yml [hunk 9]
```diff
@@ -1,25 +1,26 @@
...
     steps:
-    - uses: actions/checkout@v3
-    - run: npm ci
-    - run: npm test
-
-  # test action works running from the graph
-  test:
+      - uses: actions/checkout@v3
+      - run: |
+          npm install
```
Comment (rlch) [inline .github/workflows/test.yml]: this should be npm ci

File: .github/workflows/test.yml [hunk 10]
```diff
@@ -1,25 +1,26 @@
...
+      - run: |
+          npm run all
```
Comment (rlch) [inline .github/workflows/test.yml]: why? also |\n is not necessary

File: package.json [hunk 11]
```diff
@@ -1,36 +1,46 @@
 {
-  "name": "javascript-action",
-  "version": "1.0.0",
-  "description": "JavaScript Action Template",
-  "main": "index.js",
+  "name": "typescript-action",
+  "version": "0.0.0",
+  "private": true,
+  "description": "TypeScript template action",
```
Comment (rlch) [inline package.json]: copy name and description from workflow

File: src/core/webhook_state.ts [hunk 12]
```diff
@@ -0,0 +1,41 @@
+import axios from 'axios'
+import * as Types from '../types'
+
+function MapPRStateToNotionState(event: Types.GithubEvent): Types.NotionState {
```
Comment (rlch) [inline src/core/webhook_state.ts]: typescript functions should be lower camel-cased

File: src/core/webhook_state.ts [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+  // unimplemented
```
Comment (rlch) [inline src/core/webhook_state.ts]: wdym yunimplemented

File: src/main.ts [hunk 14]
```diff
@@ -0,0 +1,43 @@
+import * as core from '@actions/core'
+import * as github from '@actions/github'
+import * as Types from './types'
```
Comment (rlch) [inline src/main.ts]: import types explicitly rather than alias

**Review by rlch — CHANGES_REQUESTED** [review]

File: README.md [hunk 15]
```diff
@@ -53,53 +46,68 @@ See the [documentation](https://help.github.com/en/articles/metadata-syntax-for-
...
 
-## Validate
+## Usage
 
-You can now validate the action by referencing `./` in a workflow in your repo (see [test.yml](.github/workflows/test.yml))
+You can now consume the action by referencing the v1 branch
 
 ```yaml
-uses: ./
+uses: actions/javascript-action@v1
```
Comment (rlch) [inline README.md]: see prior comment, shouldn't this point to the repository?

**Review by rlch — CHANGES_REQUESTED** [review]

File: README.md (hunk 15 above)
Comment (subhroacharjee) [reply README.md]: have added the use case of this action.

**Review by rlch — APPROVED** [review]

**Pull request closed without merging**